lib/PuppeteerSharp/Browser.cs

```diff
@@ -330,19 +330,28 @@ internal async Task DisposeContextAsync(string contextId)
 
         private async void Connect_MessageReceived(object sender, MessageEventArgs e)
         {
-            switch (e.MessageID)
+            try
             {
-                case "Target.targetCreated":
-                    await CreateTargetAsync(e.MessageData.ToObject<TargetCreatedResponse>()).ConfigureAwait(false);
-                    return;
+                switch (e.MessageID)
+                {
+                    case "Target.targetCreated":
+                        await CreateTargetAsync(e.MessageData.ToObject<TargetCreatedResponse>()).ConfigureAwait(false);
+                        return;
 
-                case "Target.targetDestroyed":
-                    await DestroyTargetAsync(e.MessageData.ToObject<TargetDestroyedResponse>()).ConfigureAwait(false);
-                    return;
+                    case "Target.targetDestroyed":
+                        await DestroyTargetAsync(e.MessageData.ToObject<TargetDestroyedResponse>()).ConfigureAwait(false);
+                        return;
 
-                case "Target.targetInfoChanged":
-                    ChangeTargetInfo(e.MessageData.ToObject<TargetCreatedResponse>());
-                    return;
+                    case "Target.targetInfoChanged":
+                        ChangeTargetInfo(e.MessageData.ToObject<TargetCreatedResponse>());
+                        return;
+                }
+            }
+            catch (Exception ex)
```

Comment:
maybe we can have a utility method somewhere that will contain the switch case? we have this in each XXX_MessageReceived method

something like this:
```cs
public static void OnMessageReceived(IReadOnlyDictionary<string, Action<JToken>> messagesHandler, MessageEventArgs e)
{
    try
    {
        if (messagesHandler.TryGetValue(e.MessageID, out var handler))
        {
            handler?.Invoke(e.MessageData);
        }
    }
    catch (Exception ex)
    {
        var message = $"Browser failed to process {e.MessageID}. {ex.Message}. {ex.StackTrace}";
        _logger.LogError(ex, message);
        Connection.Close(message);
        throw;
    }
}

// usage:
OnMessageReceived(new Dictionary<string, Action<JToken>>
{
    ["Target.targetCreated"] = async data => await CreateTargetAsync(e.MessageData.ToObject<TargetCreatedResponse>()).ConfigureAwait(false),
    ["Target.targetDestroyed"] = async data => await DestroyTargetAsync(e.MessageData.ToObject<TargetDestroyedResponse>()).ConfigureAwait(false),
    ["Target.targetInfoChanged"] = data => ChangeTargetInfo(e.MessageData.ToObject<TargetCreatedResponse>()),
}, e);
```